src/Helper/ObjectParser.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Helper;
+
+use JetBrains\PhpStorm\ArrayShape;
+use JetBrains\PhpStorm\ExpectedValues;
+use ReflectionAttribute;
+use ReflectionObject;
+use ReflectionProperty;
+use Yiisoft\Validator\AfterInitAttributeEventInterface;
+use Yiisoft\Validator\RuleInterface;
+
+use function array_key_exists;
+
+final class ObjectParser
+{
+    /**
+     * @var array<string, array<string, array>>
+     */
+    #[ArrayShape([
+        [
+            'rules' => 'array',
+            'reflectionAttributes' => 'array',
+        ],
+    ])]
+    private static array $cache = [];
+    private string|null $cacheKey;
+
+    public function __construct(
+        private object $object,
+        private int $propertyVisibility = ReflectionProperty::IS_PRIVATE |
+        ReflectionProperty::IS_PROTECTED |
+        ReflectionProperty::IS_PUBLIC,
+        private bool $skipStaticProperties = false,
+        bool $useCache = true
```

Comment:
1 escaped mutant is left (changing default value from `true` to `false`). Need to adjust / actualize`ObjectDataSetTest::testCaching()` because the technique used there no longer works.

Reply:
done